src/ORM/DataObject.php

```diff
@@ -1219,7 +1221,7 @@ public function merge($rightObj, $priority = 'right', $includeRelations = true,
                     $leftComponents = $leftObj->getComponents($relationship);
                     $rightComponents = $rightObj->getComponents($relationship);
                     if ($rightComponents && $rightComponents->exists()) {
-                        $leftComponents->addMany($rightComponents->column('ID'));
+                        $leftComponents->addMany($rightComponents->sort(null)->column('ID'));
```

Comment:
```suggestion
                        $leftComponents->addMany($rightComponents->column('ID'));
```
Same reasons as above

Reply:
Discussion for this will be in https://github.com/silverstripe/silverstripe-framework/pull/11834#discussion_r2299680414 unless something about this specific method call differs substantially from that one.